### PR TITLE
Add checkstyle, findbugs and pmd checks

### PR DESCRIPTION
**Pull request by frecar**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -103,32 +106,48 @@
         <dependency>
             <groupId>org.neo4j</groupId>
             <artifactId>neo4j</artifactId>
-            <version>2.2.1</version>
+            <version>2.3.0-M02</version>
```
Comment (egabancho) [inline pom.xml]: I wouldn't make version changes within the same commit, it makes harder to debug possible problem afterwards

**Comment by egabancho** [comment]
I guess this is a work in progress but, remember to add the closes #12 to the commit message so the issue gets automatically closed

File: pom.xml (hunk 1 above)
Comment (frecar) [inline pom.xml]: I agree, we can wait with that.

**Comment by frecar** [comment]
@egabancho of course! :)

**Comment by frecar** [comment]
@egabancho @ddaze please review

@ddaze I think you want to merge this before even thinking about writing anything new (it will be a mess)  :)

**Comment by ddaze** [comment]
Looks good :+1: 
But please put the version changes in an separate commit.

**Comment by frecar** [comment]
Are we good to merge? @egabancho @ddaze

**Comment by egabancho** [comment]
I would have change a bit the commit messages, i.e. remove dot at the end summary line or add a bit more information in 6033d7b

My Java is a bit rusty right now, I'm working on it ;-), but at least next time I would definitely wait for at least one review and would put **explicitly** `Reviewed-by` in the commit message. It will help a lot in the future (talking from experience)

**Pull request merged**